### DISCUSSION TITLE
osu! can't handle paths correctly when importing a beatmap

### DISCUSSION THREAD

**Discussion by DavidHusicka** [ppy/osu#15058] [Q&A] [answered]
When importing a beatmap from a file, osu! shows error that it failed to import a beatmap showing a wrong file name. It seems like it can't handle spaces properly.
![image](https://user-images.githubusercontent.com/25853767/136955440-dc15d3c0-a98f-4fad-8339-87c3540aa0f4.png)

I tried to open the file using the "Open With Other Application" option after righ-clicking on the file in Nemo file manager and using a CLI.
![option](https://user-images.githubusercontent.com/25853767/136956696-2a00a8fb-5572-4e3a-9a9a-ad2223050bb8.png)

![cli](https://user-images.githubusercontent.com/25853767/136955674-0b791bba-00b6-4314-9d2a-400888d1811c.png)

Logs:
[database.log](https://github.com/ppy/osu/files/7329887/database.log)

**Comment by bdach** [comment]
I believe a more accurate title would be ".net core cannot handle backslash-escaped whitespace in commandline args". We're not doing anything fancy with command line argument quoting. This is a windows-ism showing (on windows, you need to quote the argument). I strongly believe that if someone is to fix this, it should be upstream (i.e. the dotnet runtime itself on linux)

Although weirdly enough it also doesn't work with the appimage when quoted - this, I believe, could be intrinsic to the appimage (possibly it unquotes the argument? not sure)

**Reply by peppy** [reply]
Definitely need some standalone testing to make sure it's not something we're doing.

**Comment by peppy** [comment]
Testing on macOS, the following works correctly:

```
dotnet run --project osu.Desktop -- ~/Downloads/1511210\ Massive\ New\ Krew\ -\ HADES.osz
```

It also works directly on a built executable. This might be linux specific, or it may be something to do with the `$HOME` substitution?

**Reply by DavidHusicka** [reply]
This works on Linux as well. It is most likely a problem with the AppImage.

**Comment by DavidHusicka** [comment, marked as answer]
Solved with https://github.com/ppy/osu-deploy/pull/110
